examples/include_file_tag.rs

```diff
@@ -0,0 +1,63 @@
+extern crate yaml_rust;
+
+mod dump_yaml;
+
+use std::path::Path;
+use std::fs::File;
+use std::io::Read;
+use std::env;
+use yaml_rust::yaml;
+use yaml_rust::scanner;
+
+struct IncludeParser<'a> {
+    root: &'a Path
+}
+
+impl<'a> IncludeParser<'a> {
+    fn new(root: &'a Path) -> IncludeParser {
+        IncludeParser {
+            root: root
+        }
+    }
+}
+
+impl<'a> yaml::YamlScalarParser for IncludeParser<'a> {
+    fn parse_scalar(&self, tag: &scanner::TokenType, value: &str) -> Option<yaml::Yaml> {
+        if let scanner::TokenType::Tag(ref handle, ref suffix) = *tag {
+            if *handle == "!" && *suffix == "include" {
```

Comment:
This line implies that the model for tag parsing implemented here differs from the YAML spec. A tag handle is nothing more but a shorthand to a URI prefix. The primary tag handle `!` defaults to the URI prefix `!` but may be redefined by a `%TAG` directive, so if a tag `!include` occurs somewhere in the source, it *may* resolve to the URI `!include`, but also to `my.fancy.uri.prefix:include`. Example:

```yaml
%YAML 1.2
%TAG ! my.fancy.uri.prefix:
---
!include foo
```

For a parser to be spec-compliant, one needs to resolve all tags to URIs and then compare those URIs against whatever is expected.